app/helpers/Intersect.js

```diff
@@ -118,6 +119,24 @@ export function intersectSetIntoBins(
         // now add the exposure period to the bin
         dayBins[daysAgo] += exposurePeriod;
 
+        // Identify if this is a new intersection by checking the crossedPaths property of the current location against the infected location j
+        if (
+          !('crossedPaths' in currentLocation) ||
+          !currentLocation['crossedPaths'].some(
+            crossedPaths =>
+              crossedPaths.time == concernArray[j].time &&
+              crossedPaths.latitude == concernArray[j].latitude &&
+              crossedPaths.longitude == concernArray[j].longitude,
+          )
+        ) {
+          newIntersections = addNewIntersection(
+            localArray,
```

Comment:
You don't need the `i` inside the method, so this could take `currentLocation` as param 1